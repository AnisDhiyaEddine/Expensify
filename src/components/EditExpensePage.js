import React from "react";

const EditExpensePage = (props) => {
console.log(props.match)
    return (<div>This is from the edit expense page</div>)
}

export default EditExpensePage;
