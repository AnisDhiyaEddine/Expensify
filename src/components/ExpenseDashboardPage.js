import React from "react";


const ExpenseDashboardPage = () => (
  <div>This is from the app dashboard page</div>
);

export default ExpenseDashboardPage;
