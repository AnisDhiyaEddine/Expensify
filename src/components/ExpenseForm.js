import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";
import "react-dates/initialize";
import "react-dates/lib/css/_datepicker.css";

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: props.expense ? props.expense.description : "",
      note: props.expense ? props.expense.expense : "",
      amount: props.expense ? props.expense.amount : "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocus: false,
      error: "",
    };
  }

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({
      description,
    }));
  };

  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({
      note,
    }));
  };

  onAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      e.target.value = amount;

      this.setState(() => ({
        amount: parseFloat(amount) || 0,
      }));
    }
  };

  onDateChange = (createdAt) => {
    this.setState(() => ({
      createdAt,
    }));
  };

  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocus: focused }));
  };

  onSubmit = (e) => {
    e.preventDefault();
    if (this.state.description && this.state.amount) {
      this.props.onSubmit({
        description: this.state.description,
        amount: this.state.amount,
        note: this.state.note,
        createdAt: this.state.createdAt.valueOf(),
      });
      this.setState(() => ({
        error: "",
      }));
    } else {
      this.setState(() => ({
        error: "Please provide the description and the amount",
      }));
    }
  };
  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            type="text"
            value={this.state.amount}
            onChange={this.onAmountChange}
            placeholder="Amount"
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocus}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea
            placeholder="Add a note for your expense (Optional"
            defaultValue={this.state.note}
            onChange={this.onNoteChange}
          ></textarea>
          <button>Add Expense</button>
        </form>
      </div>
    );
  }
}
