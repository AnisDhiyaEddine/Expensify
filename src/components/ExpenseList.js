import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectExpences from "../selectors/expenses";
import ExpenseListFilter from "./ExpenseListFilter";

const ExpenseList = (props) => (
  <div>
    {<ExpenseListFilter />}
    <h1> Expenses list</h1>
    {props.expenses.map((expense) => (
      <ExpenseListItem key={expense.id} {...expense} />
    ))}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpences(state.expenses, state.filters),
  };
};

export default connect(mapStateToProps)(ExpenseList);
