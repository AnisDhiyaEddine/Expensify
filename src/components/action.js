import React from "react";

const Action = (props) => {
  return (
    <div>
      <button
        className="big_button"
        disabled={props.disable}
        onClick={props.pickOption}
      >
        What should i do
      </button>
    </div>
  );
};

export default Action;
