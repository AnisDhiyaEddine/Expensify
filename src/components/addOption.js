import React from "react";

export default class AddOption extends React.Component {
  state = {
    error: undefined,
  };

  addOption = (event) => {
    event.preventDefault();
    this.setState(() => ({ error: "" }));

    let option = event.target.elements.addOption.value;
    event.target.elements.addOption.value = "";
    let error = this.props.addOption(option);
    if (error) {
      console.log(error);
      this.setState(() => ({ error }));
    }
  };
  render() {
    return (
      <div>
        <p className="add-option-error">{this.state.error}</p>
        <form className="add-option" onSubmit={this.addOption}>
          <input className="add-option__input" name="addOption" />
          <button className="button">Add option</button>
        </form>
      </div>
    );
  }
}
