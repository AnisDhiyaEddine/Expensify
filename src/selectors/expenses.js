// Get visible expenses
import moment from "moment";
export default (expenses = [], { text, sortBy, startDate, endDate } = {}) => {
  return expenses
    .filter((expense) => {
      const startDateMatch = moment(expense.createdAt)
        ? moment(expense.createdAt).isSameOrAfter(startDate, "day")
        : true;
      const endDateMatch = moment(expense.createdAt)
        ? moment(expense.createdAt).isSameOrBefore(endDate, "day")
        : true;

      const textMatch = expense.description
        .toLowerCase()
        .includes(text.toLowerCase());

      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => {
      if (sortBy === "date") {
        return a.createdAt < b.createdAt ? 1 : -1;
      }
      if (sortBy === "amount") {
        return a.amount < b.amount ? 1 : -1;
      }
    });
};
