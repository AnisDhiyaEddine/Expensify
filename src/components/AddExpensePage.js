import React from "react";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/expenses";
import { connect } from "react-redux";

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        <h1>his is from the add expense page</h1>
        <ExpenseForm className="ExpenseForm" onSubmit={this.onSubmit} />
      </div>
    );
  }
}

let mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense)),
});
export default connect(undefined, mapDispatchToProps)(AddExpensePage);
