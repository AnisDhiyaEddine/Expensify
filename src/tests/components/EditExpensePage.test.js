import React from "react";
import { shallow } from "enzyme";
import { EditExpensePage } from "../../components/EditExpensePage";
import expenses from "../fixtures/expenses";

let wrapper, editExpense, removeExpense, history;
beforeEach(() => {
  editExpense = jest.fn();
  removeExpense = jest.fn();
  history = {
    push: jest.fn(),
  };

  wrapper = shallow(
    <EditExpensePage
      editExpense={editExpense}
      history={history}
      removeExpense={removeExpense}
      expense={expenses[0]}
    />
  );
});

test("should render EditExpensePage", () => {
  expect(wrapper).toMatchSnapshot();
});

test("handle editExpense ", () => {
  wrapper.find(".ExpenseForm").prop("onSubmit")(expenses[0]);
  expect(history.push).toHaveBeenLastCalledWith("/");
  expect(editExpense).toHaveBeenCalledWith(expenses[0].id, expenses[0]);
});

test("handle removeExpense", () => {
  wrapper.find("button").prop("onClick")(expenses[0].id);
  expect(history.push).toHaveBeenLastCalledWith("/");
  expect(removeExpense).toHaveBeenLastCalledWith(expenses[0].id);
});
