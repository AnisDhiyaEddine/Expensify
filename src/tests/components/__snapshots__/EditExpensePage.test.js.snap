// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
<div>
  <ExpenseForm
    className="ExpenseForm"
    expense={
      Object {
        "amount": 10,
        "createdAt": 0,
        "description": "Hello",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
